code-challenges.test.js

```diff
@@ -55,6 +94,31 @@ var numbersToAdd2 = [0, 7, -8, 12]
 var numbersToAdd3 = []
 // Expected output: []
 
+describe('takes each number of an array and adding it to the next number and outputting it', () => {
+    test('if each number added to the next will equal to the next item in the output', ()=>{
+        expect(addNums(numbersToAdd1)).toEqual([2, 6, 51, 60])
+        expect(addNums(numbersToAdd2)).toEqual([0, 7, -8, 12])
+        expect(addNums(numbersToAdd3)).toEqual([])
+    })
+})
 
 
 // b) Create the function that makes the test pass.
+
+// const addNums = (array) => {
+    let newArr1 = []
+    // for(let i=0; i<array.length; i++){
+        if(numbersToAdd1[0] === "number"){
+           return array[i][0], array[i][0] + array[i][1], array[i][0] + array[i][1] + array[i][2], array[i][0] + array[i][1] + array[i][2] + array[i][3], newArr1.push(array[i])
```

Comment:
I see what you're doing here and this would make for good notes of a specific use case. But  as is it's a good example of "hard coding".  We want our code to be dynamic. 

What would you do if you were given an input of an array with 100 indexes? What about 1000? When we write our code we are solving for non specific cases. Our code should work given a wide variety of inputs.

How would you solve this issue without code just using plain language?